packages/salesforcedx-vscode-core/src/commands/forceAuthDevHub.ts

```diff
@@ -37,6 +48,49 @@ export class ForceAuthDevHubExecutor extends SfdxCommandletExecutor<{}> {
     command.withArg('--setdefaultdevhubusername');
     return command.build();
   }
+
+  public async execute(response: ContinueResponse<any>): Promise<void> {
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+
+    const execution = new CliCommandExecutor(this.build(response.data), {
+      cwd: getRootWorkspacePath()
+    }).execute(cancellationToken);
+
+    execution.processExitSubject.subscribe(async () => {
+      const globalDevHubName = await OrgAuthInfo.getDefaultDevHubUsernameOrAlias(
+        false,
+        ConfigSource.Global
+      );
+
+      if (isNullOrUndefined(globalDevHubName)) {
+        const localDevHubName = await OrgAuthInfo.getDefaultDevHubUsernameOrAlias(
+          false,
+          ConfigSource.Local
+        );
+
+        if (isNullOrUndefined(localDevHubName) === false) {
+          await this.setGlobalDefaultDevHub(String(localDevHubName));
+        }
+      }
+    });
+
+    this.attachExecution(execution, cancellationTokenSource, cancellationToken);
+  }
+
+  private async setGlobalDefaultDevHub(newUsername: string) {
+    const homeDirectory = require('os').homedir();
+    const configFileName = 'sfdx-config.json';
```

Comment:
Instead of redefining the file name you can reference it using the constant defined [here](https://github.com/forcedotcom/salesforcedx-vscode/blob/da004e6a7704915a47abcfa87a0756344eba4bac/packages/salesforcedx-vscode-core/src/constants.ts#L8)